src/main/java/org/kohsuke/github/GHRepositoryRule.java

```diff
@@ -0,0 +1,667 @@
+         * Gets the integration id.
+         *
+         * @return the integration id
+         */
+        public Integer getIntegrationId() {
+            return this.integrationId;
+        }
+    }
+
+    /**
+     * Operator parameter.
+     */
+    public static enum Operator {
+        /**
+         * starts_with
+         */
+        STARTS_WITH,
+
+        /**
+         * ends_with
+         */
+        ENDS_WITH,
+
+        /**
+         * contains
+         */
+        CONTAINS,
+
+        /**
+         * regex
+         */
+        REGEX
+    }
+
+    /**
+     * Workflow file reference parameter.
+     */
+    public static class WorkflowFileReference {
+        private String path;
+        private String ref;
+        private int repositoryId;
+        private String sha;
+
+        /**
+         * Instantiates a new workflow file reference.
+         */
+        public WorkflowFileReference() {
+        }
+
+        /**
+         * Instantiates a new workflow file reference.
+         *
+         * @param path
+         *            the path
+         * @param ref
+         *            the ref
+         * @param repositoryId
+         *            the repository id
+         * @param sha
+         *            the sha
+         */
+        public WorkflowFileReference(String path, String ref, int repositoryId, String sha) {
+            this.path = path;
+            this.ref = ref;
+            this.repositoryId = repositoryId;
+            this.sha = sha;
+        }
+
+        /**
+         * Gets the path.
+         *
+         * @return the path
+         */
+        public String getPath() {
+            return this.path;
+        }
+
+        /**
+         * Gets the ref.
+         *
+         * @return the ref
+         */
+        public String getRef() {
+            return this.ref;
+        }
+
+        /**
+         * Gets the repository id.
+         *
+         * @return the repository id
+         */
+        public int getRepositoryId() {
+            return this.repositoryId;
+        }
+
+        /**
+         * Gets the sha.
+         *
+         * @return the sha
+         */
+        public String getSha() {
+            return this.sha;
+        }
+    }
+
+    /**
+     * Code scanning tool parameter.
+     */
+    public static class CodeScanningTool {
+        private AlertsThreshold alertsThreshold;
+        private SecurityAlertsThreshold securityAlertsThreshold;
+        private String tool;
+
+        /**
+         * Instantiates a new code scanning tool.
+         */
+        public CodeScanningTool() {
+        }
+
+        /**
+         * Instantiates a new code scanning tool.
+         *
+         * @param alertsThreshold
+         *            the alerts threshold
+         * @param securityAlertsThreshold
+         *            the security alerts threshold
+         * @param tool
+         *            the tool
+         */
+        public CodeScanningTool(AlertsThreshold alertsThreshold,
```

Comment:
Jackson will do this without a public constructor.  Make it internal  or remove it entirely.